### PR TITLE
Reworked branch defines into single compatibility define

### PR DESCRIPTION
**Pull request by gruxor**
## Description
- Moved existing branch defines out of their own file and into config.h
- Single definition exists for any future compatibility checks in 3rd-party code, but old branch defines are retained for backwards compatibility with existing projects
- Doesn't really change anything functionally, but may clear up some confusion for people who never used the repo pre-merge

## **Discord contact info**
Crater#7777

### CODE REVIEW

File: include/config.h [hunk 1]
```diff
@@ -36,4 +36,14 @@
 #endif
 #endif
 
+// Compatibility definition for other projects to detect pokeemerald-expansion
+#define RHH_EXPANSION
+
+// Legacy branch-based defines included for backwards compatibility
+#ifdef RHH_EXPANSION
```
Comment (AsparagusEduardo) [inline include/config.h:43]: Don't know if this ifdef is even needed.

Comment (gruxor) [reply include/config.h:43]: Was thinking about that in retrospect, it does seem redundant. I think I was initially considering if someone wanted to "turn off" all of the defines at once without commenting them out, but that's pretty trivial to do, and I'm not sure why anyone would want to do that in the first place.

Feel free to remove the ifdef, or I can update with a commit if you want.

Comment (AsparagusEduardo) [reply include/config.h:43]: Please do.

Comment (gruxor) [reply include/config.h]: Fixed 👍

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**